src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/DeploymentOptions.cs

```diff
@@ -33,7 +41,7 @@ public class DeploymentOptions
     {
         #region Properties
 
-        public DeploymentOptionProperty<object> IgnoreTableOptions { get; set; }
+        public DeploymentOptionProperty<bool> IgnoreTableOptions { get; set; }
```

Comment:
After you merge in you should change the rest of the properties to their actual types.

Reply:
In one of my previous iterations these properties are changed to bool/string type, In runtime the property Intialization with values required type casting which ends with 'if else if' statements. I'll update them now as my final iteration(hopefully :)). thanks.

Reply:
I'm a bit confused here - what exactly are you saying is the issue? Were you able to find a way to have strong typing without the `Activator.CreateInstance` solution I have in this PR?

We shouldn't keep these types as `object` - that kind of defeats the purpose of doing generics in the first place.